src/pages/iou/IOUDetailsModal.js

```diff
@@ -77,29 +81,109 @@ class IOUDetailsModal extends Component {
     constructor(props) {
         super(props);
 
+        // We always have the option to settle manually
+        const paymentOptions = [CONST.IOU.PAYMENT_TYPE.ELSEWHERE];
+
+        // Only allow settling via PayPal.me if the submitter has a name set
+        if (props.iouReport.submitterPayPalMeAddress) {
+            paymentOptions.push(CONST.IOU.PAYMENT_TYPE.PAYPAL_ME);
+        }
+
+        this.submitterPhoneNumber = undefined;
+        this.isComponentMounted = false;
+
         this.state = {
-            // Temporarily placeholder, as the Venmo/Paypal issue will introduce settlement types.
-            settlementType: 'Elsewhere',
+            paymentType: CONST.IOU.PAYMENT_TYPE.ELSEWHERE,
+            isSettlementMenuVisible: false,
+            paymentOptions,
         };
 
         this.performIOUPayment = this.performIOUPayment.bind(this);
     }
 
     componentDidMount() {
+        this.isComponentMounted = true;
         fetchIOUReportByID(this.props.route.params.iouReportID, this.props.route.params.chatReportID);
+        this.addVenmoPaymentOptionIfAvailable();
+    }
+
+    componentWillUnmount() {
+        this.isComponentMounted = false;
+    }
+
+    setMenuVisibility(isSettlementMenuVisible) {
+        this.setState({isSettlementMenuVisible});
     }
 
     performIOUPayment() {
         payIOUReport({
             chatReportID: this.props.route.params.chatReportID,
             reportID: this.props.route.params.iouReportID,
-            paymentMethodType: this.state.settlementType,
+            paymentMethodType: this.state.paymentType,
+            amount: this.props.iouReport.total,
+            currency: this.props.iouReport.currency,
+            submitterPayPalMeAddress: this.props.iouReport.submitterPayPalMeAddress,
+            submitterPhoneNumber: this.submitterPhoneNumber,
         });
     }
 
+    /**
+     * @param {String} phoneNumber
+     * @returns {Boolean}
+     */
+    isValidUSPhone(phoneNumber) {
+        // Remove alphanumeric characters and validate that this is in fact a phone number
+        return CONST.REGEX.PHONE_E164_PLUS.test(phoneNumber.replace(CONST.REGEX.NON_ALPHA_NUMERIC, ''))
+
+            // Next make sure it's a US phone number
+            && CONST.REGEX.US_PHONE.test(phoneNumber);
+    }
+
+    /**
+     * Checks to see if we can use Venmo. The following conditions must be met:
+     *
+     *   1. The IOU report currency is USD
+     *   2. The submitter has as a valid US phone number
+     *   3. Venmo app is installed
+     *
+     */
+    addVenmoPaymentOptionIfAvailable() {
+        if (this.props.iouReport.currency !== CONST.CURRENCY.USD) {
+            return;
+        }
+
+        const submitterPhoneNumbers = lodashGet(this.props.iouReport, 'submitterPhoneNumbers', []);
+
+        if (_.isEmpty(submitterPhoneNumbers)) {
```

Comment:
NAB: Is there any benefit to the newline here? Personally, I think it made this function a little trickier to read.

Reply:
>I think it made this function a little trickier to read.

That's interesting. I put visual distance before `if` statements sometimes because having them less compressed makes them slightly clearer to me. Will change it thought if there's a preference to not do this.

Reply:
I guess it's just personal taste. I think it stood out to me this time because of the multiple conditions. If it was just one I probably wouldn't have even noticed it.